cmake/z3_add_component.cmake

```diff
@@ -100,6 +100,11 @@ macro(z3_add_component component_name)
   # Note: We don't check the sources exist here because
   # they might be generated files that don't exist yet.
 
+  if(Z3_SINGLE_THREADED)
```

Comment:
I think it will be best to push this to the files themselves and outside of the build system